src/openms/source/ANALYSIS/ID/SiriusAdapterAlgorithm.cpp

```diff
@@ -183,7 +184,31 @@ namespace OpenMS
     {
       return tmp_ms_file_;
     }
-    
+
+    std::vector<String> SiriusAdapterAlgorithm::sortSiriusWorkspacePathsByScanIndex(const std::vector<String>& siriusworkspacepaths)
+    {
+      std::vector<String> sorted_siriusworkspacepaths;
+      std::vector<SiriusAdapterAlgorithm::SiriusWorkspaceIndex> indices;
+
+      for (size_t i = 0; i < siriusworkspacepaths.size(); i++)
+      {
+        SiriusAdapterAlgorithm::SiriusWorkspaceIndex wsi;
+        wsi.array_index = i;
+        wsi.scan_index = SiriusMzTabWriter::extractScanIndex(siriusworkspacepaths[i]);
+        indices.emplace_back(wsi);
+      }
+
+      std::sort(indices.begin(), indices.end(), [](const SiriusAdapterAlgorithm::SiriusWorkspaceIndex& i, const SiriusAdapterAlgorithm::SiriusWorkspaceIndex& j) { return i.scan_index < j.scan_index; } );
+
+      std::vector<String> sorted_subdirs;
+      for (const auto& index : indices)
+      {
+        sorted_siriusworkspacepaths.emplace_back(siriusworkspacepaths[index.array_index]);
```

Comment:
emplace_back is misleading, because this is a copy, not a move.
It cannot be a move since `siriusworkspacepaths` is const&. If you pass it in a rvalue-ref (`&&`) you can move from it. Will be faster, but its probably not that important.

Reply:
Just to be sure, since the initial `subdirs` vector is not needed anymore after calling the sort function to generate the `sorted_subdirs`, the initial `subdirs` vector can be used as a rvalue to perform move instead of copy.

Declaration: 
`std::vector<String> SiriusAdapterAlgorithm::sortSiriusWorkspacePathsByScanIndex(std::vector<String>&& subdirs);`

Call in e.g. SiriusAdapter.cpp
`vector<String> sorted_subdirs = SiriusAdapterAlgorithm::sortSiriusWorkspacePathsByScanIndex(std::move(subdirs));`

To call the function with move schematics, `std:move()` has to be used to allow the lvalue "subdirs" to be used as a rvalue for the move. 

Or is there more to it?

Reply:
@cbielow Unfortunately it seems that the rvalue reference can not be wrapped in python via autowrap.

Reply:
https://github.com/uweschmitt/autowrap/issues/95